core/src/main/java/org/apache/calcite/avatica/remote/AvaticaCommonsHttpClientImpl.java

```diff
@@ -127,11 +121,54 @@ private void initializeClient() {
     final String maxCnxnsPerRoute = System.getProperty(MAX_POOLED_CONNECTION_PER_ROUTE_KEY,
         MAX_POOLED_CONNECTION_PER_ROUTE_DEFAULT);
     pool.setDefaultMaxPerRoute(Integer.parseInt(maxCnxnsPerRoute));
+  }
 
-    this.authCache = new BasicAuthCache();
+  protected Registry<ConnectionSocketFactory> configureSocketFactories() {
+    RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+    configureHttpRegistry(registryBuilder);
+    configureHttpsRegistry(registryBuilder);
+    return registryBuilder.build();
+  }
 
-    // A single thread-safe HttpClient, pooling connections via the ConnectionManager
-    this.client = HttpClients.custom().setConnectionManager(pool).build();
+  protected void configureHttpsRegistry(RegistryBuilder<ConnectionSocketFactory> registryBuilder) {
+    if (!configureHttpsSocket) {
```

Comment:
What about looking at the scheme of the URL instead of using "one of" the TLS configuration options?

Reply:
Actually, if we look at the URL, then may push this check up to `configureSocketFactories()` and call only one of `configureHttpRegsitry` and `configureHttpsRegistry`?

Reply:
I agree, but sometimes various services can have varied requirements. The best and simple case is to have either http or https, however I didn't want to change the original implementation, so I added both of them here as well.
What do you think?

Reply:
I think there's a strong coupling of HttpClient to a specific avatica server instance ( going off of memory). If that's the case, then there's no problem in making the determination as to what kind of http client we need to set up.

Reply:
I looked through the code. That's correct. The `HttpHost` required by `CloseableHttpClient` is created in the constructor itself. `HttpHost` is the one that uses the URL provided during the initialization. So there's a tight coupling here and there will never be a case where both of them will be together. I will update the code to configure either of these factories depending on URL prefix as `http` vs `https`. What do you think?

Reply:
Actually I will use `url.getProtocol()` method to handle it for me. Seems better option.

Reply:
As FYI, I ended up using `host.getSchemeName()` where host is a `HttpPost` object. The value is populated by `url.getProtocol()` only.